### PR TITLE
[WIP] Travis: Deploy Debian packages to Bintray

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**
After a successful build of a tag Travis will push packages to bintray. Even when there is native support to deploy to Bintray in Travis, we don't use it because it's too complicated, you have to write almost one json file per file to upload so we can't just say upload everything like `deb/*.deb`.

Upload credentials are expected to be defined in Travis repository environment variables: `BINTRAY_USER` and `BINTRAY_API_KEY`.

If the tag version contains a `-`, then the release is assumed to be a pre-release (as per SemVer specs), so it is uploaded to a "prerelease" component. All other tags are uploaded to the "release" component.

The Bintray apt repository is located at https://bintray.com/sociomantic-tsunami/ebtree.

To add it to the apt sources:
```
apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 379CE192D401AB61
echo "deb https://dl.bintray.com/sociomantic-tsunami/ebtree $(lsb_release -cs) release prerelease" | \
	sudo tee -a /etc/apt/sources.list.d/ebtree.list
cat <<EOT | tee -a /etc/apt/preferences.d/ebtree
Package: *
Pin: release o=Bintray,a=xenial,c=release
Pin-Priority: 600

Package: *
Pin: release o=Bintray,a=xenial,c=prerelease
Pin-Priority: 40
EOT
sudo apt-get update
```

This way pre-releases will only be installed if the version is specified explicitly. If you only want the regular releases, just remove the `prerelease` from the first `echo` and forget about the `/etc/apt/preferences.d/ebtree` file.

I tested this in my fork and it seems to work well when pushing a tag, we can try it out after it's merged to produce a new v6.0.socio7 release candidate (-rc2).

The idea is to use this scheme in all repos that produce debian packages.

### CODE REVIEW

File: Dockerfile.trusty [hunk 1]
```diff
@@ -1,3 +1,3 @@
-FROM sociomantictsunami/base:v1-trusty
+FROM sociomantictsunami/base:v2test-trusty
```
Comment (leandro-lucarella-sociomantic) [inline Dockerfile.trusty:1]: We should tag this cachalot docker image as the final `v2` before merging this.

**Comment by mihails-strasuns-sociomantic** [comment]
Do you intend to move these scripts to cachalot? I would like to keep propagation of all those CI scripts in individual repos as contained as possible.

**Comment by leandro-lucarella-sociomantic** [comment]
> Do you intend to move these scripts to cachalot? I would like to keep propagation of all those CI scripts in individual repos as contained as possible.

To beaver. Maybe we could even not merge this PR, as I'm close to an initial beaver PR covering this use.

**Comment by mihails-strasuns-sociomantic** [comment]
Now that we have both cahalot, beaver and dmd-transitional repos at once and nothing is automated yet, I am very confused which things go where :(

**Comment by leandro-lucarella-sociomantic** [comment]
* cachalot only for building docker images
* beaver only for helper travis scripts (to be added as a submodule in all other projects)
* dmd-transitional will use beaver to make packages and upload them to bintray
* same for ebtree and all the other projects that we are now built in cachalot as part of the image creation, all those will be now built, packaged and uploaded to the debian repo by the respective repos as part of the regular travis builds (although packages will only be uploaded for tagged builds) and then just `apt-get install`ed as part of cachalot docker image creation (as we used to do internally)

For now I was just playing around with different repos to "create" beaver, for which I will have a PR soon (it's just being hard for me to find time to complete it). After that all should be much more clear.

**Comment by leandro-lucarella-sociomantic** [comment]
I'm sorry for the mess, I ACK that for now unless you can get into my head, it might be quite confusing...

**Comment by leandro-lucarella-sociomantic** [comment]
Closing in favour of #9.

**Pull request closed without merging**